pyccel/ast/core.py

```diff
@@ -2537,6 +2577,108 @@ def __str__(self):
     def is_unused(self):
         return False
 
+class InlineFunctionDef(FunctionDef):
+    """
+    Represents a function definition for an inline function.
+
+    Parameters
+    ----------
+    See FunctionDef
+
+    namespace_imports : Scope
+                        The objects in the scope which are available due to imports
+    """
+    __slots__ = ('_namespace_imports','_orig_args','_new_args','_new_local_vars', '_if_block_replacements')
+
+    def __init__(self, *args, namespace_imports = None, **kwargs):
+        self._namespace_imports = namespace_imports
+        super().__init__(*args, **kwargs)
+        self._orig_args = tuple(a.var for a in self.arguments)
+        self._new_args  = None
+        self._new_local_vars = None
+        self._if_block_replacements = None
+
+    @property
+    def is_inline(self):
+        """ True if the function should be printed inline """
+        return True
+
+    @property
+    def namespace_imports(self):
+        """ The objects in the scope which are available due to imports
+        """
+        return self._namespace_imports
+
+    def swap_in_args(self, args, new_local_vars):
+        """ Modify the body of the function by replacing the argument and local Variables
+        with the provided argument and local Variables
+        """
+        assert self._new_args is None
+        assert self._new_local_vars is None
+        # Collect the function arguments and the expressions they will be replaced with
+        self._new_local_vars = tuple(new_local_vars)
+
+        # We cannot replace with singletons as this cannot be reversed
+        self._new_args  = tuple(NilArgument() if a is Nil() else \
+                        PyccelAssociativeParenthesis(a) if isinstance(a, PyccelOperator) \
+                        else a for a in args)
+
+        # Replace the arguments in the code
+        self.body.substitute(self._orig_args+self.local_vars, self._new_args+self._new_local_vars, invalidate=False)
+
+    def swap_out_args(self):
+        """ Modify the body of the function by reinstating the original argument and local Variables
+        """
+        self.body.substitute(self._new_args+self._new_local_vars, self._orig_args+self.local_vars, invalidate=False)
+        self._new_args = None
+        self._new_local_vars = None
```

Comment:
No. I was not sure how to handle inline functions in the python printer. In the end I decided to leave them as they are. The "translated" code is therefore more or less identical to the original code and the inline is only signalled through the inline decorator. If you think it would be better to show the inlined version then I can add a `_handle_inline_func_call` method to the python printer which would call this method

Reply:
No need to change the Python printer. I was only curious to know where you use this function...?

Reply:
It is used in the `_handle_inline_func_call` function in the Fortran and the C printers